docs: update smart contracts async flow diagram

VolodymyrBg opened the pull request:
## Proposed changes

1. Updated the smart_contracts_async_flow.svg diagram in the custom contracts tutorial to:
- Add the Blockchain Connector component to accurately represent the architecture
- Improve the flow visualization with more detailed steps
- Add color coding for better visual distinction between components
- Include a note about event subscription options
- Make the diagram more comprehensive and modern
2. Removed the TODO comment from the custom_contracts/index.md file

Fixes # <issue_number_here>

<hr>

## Types of changes

<!-- to mark a point done, place an x in square brackets. eg [x]-->
<!-- - [x] done with this task-->
<!----Please delete options that are not relevant. And in order to tick the check box just but x inside them for example [x] like this----->

- [ ] Bug fix 
- [ ] New feature added
- [x] Documentation Update 

<hr>

## Please make sure to follow these points 

<!-- to mark a point done, place an x in square brackets. eg [x]-->
<!-- - [x] done with this task-->
<!----Please delete options that are not relevant. And in order to tick the check box just but x inside them for example [x] like this----->

- [x] I have read the contributing guidelines.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes have sufficient code coverage (unit, integration, e2e tests).

<hr>

## Screenshots (If Applicable)

<hr>


## Other Information

The previous diagram was simplified and didn't accurately represent the role of the Blockchain Connector or the complete flow of events. This update provides a more accurate and detailed representation of how custom contract interactions work in FireFly, which will help developers better understand the asynchronous nature of blockchain transactions.

VolodymyrBg committed:
Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update index.md

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

EnriqueL8 reviewed (commented):
Thanks @VolodymyrBg this is super useful, it seems that the SVG is cut and I can't see the store events flow and emit events to the App, could you please add that?

EnriqueL8 commented on doc-site/docs/images/smart_contracts_async_flow.svg:
A few things, 
- the `forward transaction` from the connector to the blockchain should be `submit transaction` as it's in the blockchain connector where the transaction is signed and submitted using the specific DLT format
- This `Accepted` flow is not strictly correct and depends if it's the TX has been submitted synchronously through the app using `confirm=true` or not. You only get a "Submitted" event to the app not `Accepted`
- `Process Transaction` - > `Perform Transaction` 
- `Forward events` -> `Forward interested blockchain events`

EnriqueL8 reviewed (commented):
Thanks , looking better! A few changes needed please

VolodymyrBg commented:
@EnriqueL8 Corrected. I can squash commits if it's necessary, cause too many of them. Sorry for mess

EnriqueL8 commented:
No worries, could you squash please :)

VolodymyrBg commented:
@EnriqueL8 Done

EnriqueL8 committed:
Merge branch 'main' into BG

EnriqueL8 commented on doc-site/docs/images/smart_contracts_async_flow.svg:
After reviewing the code flow and where the TODO was, this diagram doesn't show the Async behavior of a smart contract invoke.  The blockchain connector will return a transaction accepted before it's submitted to the blockchain and the response of the operation being completed is a receipt from the blockchain so two things need changing:
- `Transaction accepted` from connector to FireFly should just after the `Submit Transaction` 
- `Before the emit blockchain events` there is also a `Retrieve receipt/confirmation for TX` and then `confirm tx to FireFly`. At this point the operation is updated.

EnriqueL8 reviewed (commented):
Sorry for spotting this later in the review

VolodymyrBg committed:
Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

VolodymyrBg commented:
@EnriqueL8 Hope now it's ok

EnriqueL8 committed:
Merge branch 'main' into BG

VolodymyrBg committed:
Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

Update smart_contracts_async_flow.svg

Signed-off-by: VolodymyrBg <aqdrgg19@gmail.com>

EnriqueL8 reviewed (approved):
Awesome thanks @VolodymyrBg!
